api/krusty/remoteloader_test.go

```diff
@@ -0,0 +1,357 @@
+	//   with-submodule.git/	- includes `simple` as a submodule
+	//     submodule/			- the submodule referencing `simple
+	createGitRepos := func(t *testing.T) testRepos {
+		t.Helper()
+
+		bash := func(script string) {
+			cmd := exec.Command("sh", "-c", script)
+			o, err := cmd.CombinedOutput()
+			if err != nil {
+				t.Fatalf("error running %v\nerr: %v\n%s", script, err, string(o))
+			}
+		}
+		root := t.TempDir()
+		bash(fmt.Sprintf(`
+set -eux
+
+export ROOT="%s"
+export GIT_AUTHOR_EMAIL=nobody@kustomize.io
+export GIT_AUTHOR_NAME=Nobody
+export GIT_COMMITTER_EMAIL=nobody@kustomize.io
+export GIT_COMMITTER_NAME=Nobody
+
+cp -r testdata/remoteload/simple $ROOT/simple.git
+(
+	cd $ROOT/simple.git
+	git init --initial-branch=main
+	git add .
+	git commit -m "import"
+	git checkout -b change-image
+	cat >>kustomization.yaml <<EOF
+
+images:
+- name: nginx
+  newName: nginx
+  newTag: "2"
+EOF
+	git commit -am "image change"
+	git checkout main
+)
+cp -r testdata/remoteload/multibase $ROOT/multibase.git
+(
+	cd $ROOT/multibase.git
+	git init --initial-branch=main
+	git add .
+	git commit -m "import"
+)
+(
+	mkdir $ROOT/with-submodule.git
+	cd $ROOT/with-submodule.git
+	git init --initial-branch=main
+	git submodule add $ROOT/simple.git submodule
+	git commit -m "import"
+)
+`, root))
+		return testRepos{
+			root:          root,
+			simple:        "simple.git",
+			multiBaseDev:  "multibase.git",
+			withSubmodule: "with-submodule.git",
+		}
+	}
+
+	const simpleBuild = `apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    app: myapp
+  name: myapp-pod
+spec:
+  containers:
+  - image: nginx:1.7.9
+    name: nginx
+`
+	var multibaseDevExampleBuild = strings.ReplaceAll(simpleBuild, "myapp-pod", "dev-myapp-pod")
+
+	repos := createGitRepos(t)
+	tests := []struct {
+		name          string
+		kustomization string
+		expected      string
+		err           string
+		skip          bool
+	}{
+		{
+			name: "simple",
+			kustomization: `
+resources:
+- file://$ROOT/simple.git
+`,
+			expected: simpleBuild,
+		},
+		{
+			name: "has path",
+			kustomization: `
+resources:
+- file://$ROOT/multibase.git/dev
+`,
+			expected: multibaseDevExampleBuild,
+		},
+		{
+			name: "has ref",
+			kustomization: `
+resources: 
+- "file://$ROOT/simple.git?ref=change-image"
+`,
+
+			expected: strings.ReplaceAll(simpleBuild, "nginx:1.7.9", "nginx:2"),
+		},
+		{
+			// Version is the same as ref
+			name: "has version",
+			kustomization: `
+resources:
+- file://$ROOT/simple.git?version=change-image
+`,
+			expected: strings.ReplaceAll(simpleBuild, "nginx:1.7.9", "nginx:2"),
+		},
+		{
+			name: "has submodule",
+			kustomization: `
+resources:
+- file://$ROOT/with-submodule.git/submodule
+`,
+			expected: simpleBuild,
+		},
+		{
+			name: "has submodule but not initialized",
+			kustomization: `
+resources:
+- file://$ROOT/with-submodule.git/submodule?submodules=0
+`,
+			err: "unable to find",
+		},
+		{
+			name: "has origin annotation",
```

Comment:
@natasha41575 we previously had [both `TestRemoteResourceAnnoOrigin` and `TestRemoteResourceAsBaseWithAnnoOrigin`](https://github.com/kubernetes-sigs/kustomize/blob/23de1499c27b1d040e90c8094051fe960661196a/api/krusty/remoteload_test.go#L360-L434), where the latter looks like it just adds an overlay over the same remote-resources Kustomization. That seems like the same thing to me from a remote resources perspective. Am I missing something we should continue to cover?

Reply:
From the remote source perspective, it's the same. I added both to ensure that the origin annotations were preserved throughout the overlays, but there are a few other tests that also cover that behavior, so I'm okay to remove coverage for `TestRemoteResourceAsBaseWithAnnoOrigin` so long as we still have coverage for `TestRemoteResourceAnnoOrigin`.